larpandoracontent/LArHelpers/LArPfoHelper.cc

```diff
@@ -485,6 +495,48 @@ const Vertex *LArPfoHelper::GetVertex(const ParticleFlowObject *const pPfo)
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+const Vertex *LArPfoHelper::GetTestBeamInteractionVertex(const ParticleFlowObject *const pPfo)
+{
+    if (pPfo->GetVertexList().empty() || !pPfo->GetParentPfoList().empty() || !LArPfoHelper::IsTestBeam(pPfo))
+        throw StatusCodeException(STATUS_CODE_NOT_FOUND);
+
+    const Vertex *pInteractionVertex(LArPfoHelper::GetVertexType(pPfo->GetVertexList(), VERTEX_INTERACTION));
+
+    if (VERTEX_3D != pInteractionVertex->GetVertexType())
+        throw StatusCodeException(STATUS_CODE_FAILURE);
+
+    return pInteractionVertex;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+const Vertex *LArPfoHelper::GetVertexType(const VertexList &vertexList, const VertexLabel &vertexLabel)
```

Comment:
Better to call this `GetVertexOfLabel` or `GetVertexWithLabel`, or something that communicates that we're accessing via VertexLabel, rather than via VertexType (our two distinguishing enumerated types).

Worry what happens if there's multiple vertices with the same label in the list. Here you'll get the last vertex with that label. Is that OK? No current use-case of course, but should be prepared!

VertexLabel argument should be by value, rather than reference (it's just an enumerated type).

Reply:
Hi John,  Good point about the naming scheme.  I will update that.  If there's multiple vertices of the same time in the list passed to this function it throws an exception rather than returning the last two.  I made that decision as I don't think there should ever be a case where a particle has multiple types of the same vertex.  Will also remove the reference too as you're right it doesn't save any overheads.